zerver/lib/actions.py

```diff
@@ -3383,14 +3383,22 @@ def notify_realm_filters(realm):
 #   * Named groups will be converted to numbered groups automatically
 #   * Inline-regex flags will be stripped, and where possible translated to RegExp-wide flags
 def do_add_realm_filter(realm, pattern, url_format_string):
-    # type: (Realm, text_type, text_type) -> None
-    RealmFilter(realm=realm, pattern=pattern,
-                url_format_string=url_format_string).save()
+    # type: (Realm, text_type, text_type) -> int
+    realm_filter = RealmFilter(
+        realm=realm, pattern=pattern,
+        url_format_string=url_format_string)
+    realm_filter.full_clean()
+    realm_filter.save()
     notify_realm_filters(realm)
 
-def do_remove_realm_filter(realm, pattern):
-    # type: (Realm, text_type) -> None
-    RealmFilter.objects.get(realm=realm, pattern=pattern).delete()
+    return realm_filter.id
+
+def do_remove_realm_filter(realm, pattern=None, id=None):
+    # type: (Realm, text_type, int) -> None
+    if pattern:
```

Comment:
use `is not None` stylistically.  Also, in mypy one should use `Optional[text_type], Optional[int]` for the other arguments.